### DISCUSSION TITLE
body-bg overrides all components background

### DISCUSSION THREAD

**Discussion by amolitos** [twbs/bootstrap#38733] [Q&A]
In a new project, using bootstrap 5.3.0 and SASS, I give a value to the variable $body-bg

![Captura de Pantalla 2023-06-08 a la(s) 21 31 16](https://github.com/twbs/bootstrap/assets/15843598/c6cc3d4a-4248-4788-918d-7756ac11c9ea)

But this causes all components to inherit the color from the $body-bg variable.

![Captura de Pantalla 2023-06-08 a la(s) 21 22 55](https://github.com/twbs/bootstrap/assets/15843598/5f016a35-9e2b-40d7-9ea1-537566964a25)
![Captura de Pantalla 2023-06-08 a la(s) 21 40 43](https://github.com/twbs/bootstrap/assets/15843598/df60106d-14b1-484e-9cad-ae37c4709d18)
![Captura de Pantalla 2023-06-08 a la(s) 21 44 12](https://github.com/twbs/bootstrap/assets/15843598/2d2f6a4c-d533-48ed-88e3-987955b49841)

Shouldn't it just set the background color of the body? How can I solve that?

**Comment by julien-deramond** [comment]
`$body-bg` defines the body background color but is also reused to define the default background color of most of the components too. Think of it as the default background color for everything.
From that, you can override specifically the other background colors like `$dropdown-bg`, `$accordion-bg`, `$offcanvas-bg-color`, etc.
If you want only to change the background color of the body, you probably just need to apply a `.bg-danger` (for example)  to the `<body>` HTML element.

**Reply by alexandergitter** [reply]
This surprised me twice in different projects when upgrading from Bootstrap 5.2 to 5.3 (e.g. `$card-bg` was hardcoded to `white` before 5.3).
I think it would be useful to mention this change in the migration guide (https://getbootstrap.com/docs/5.3/migration)

**Reply by julien-deramond** [reply]
Thanks for the feedback @alexandergitter. Add it to my TODO to see how to improve the migration guide regarding this modification.

**Comment by ray73864** [comment]
This has caught me out too.

I've set the 'style' on the ```<body>```, but it also looks like I need to override the ```$table-bg``` too, since in 5.2, setting ```$body-bg``` affected the body and tables, but didn't affect cards, etc...

In 5.3, setting ```$body-bg``` is affecting all of them.

I only want the overall body and tables to have the same background, while everything else should be white.
